MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs

```diff
@@ -32,340 +30,299 @@ namespace MediaBrowser.MediaEncoding.Encoder
     public class MediaEncoder : IMediaEncoder, IDisposable
     {
         /// <summary>
-        /// The _logger
-        /// </summary>
-        private readonly ILogger _logger;
-
-        /// <summary>
-        /// Gets the json serializer.
+        /// Gets the encoder path.
         /// </summary>
-        /// <value>The json serializer.</value>
-        private readonly IJsonSerializer _jsonSerializer;
+        /// <value>The encoder path.</value>
+        public string EncoderPath => FFmpegPath;
 
         /// <summary>
-        /// The _thumbnail resource pool
+        /// The location of the discovered FFmpeg tool.
         /// </summary>
-        private readonly SemaphoreSlim _thumbnailResourcePool = new SemaphoreSlim(1, 1);
-
-        public string FFMpegPath { get; private set; }
+        public FFmpegLocation EncoderLocation { get; private set; }
 
-        public string FFProbePath { get; private set; }
+        private FFmpegLocation ProbeLocation { get; set; }
```

Comment:
Why is this a property instead of a field?

Reply:
Probably because it was originally 'Public' and got changed to 'Private'.  Shall change to field.